pkg/testsuite/mock/blockissuer.go

```diff
@@ -47,101 +43,109 @@ type BlockIssuer struct {
-	privateKey ed25519.PrivateKey
+	keyManager *wallet.KeyManager
+	Client     Client
+	// LatestBlockIssuanceResp is the cached response from the latest query to the block issuance endpoint.
+	latestBlockIssuanceResp   *api.IssuanceBlockHeaderResponse
+	blockIssuanceResponseUsed bool
+	mutex                     syncutils.RWMutex
 
-	events *Events
-
-	workerPool *workerpool.WorkerPool
-
-	optsTipSelectionTimeout       time.Duration
-	optsTipSelectionRetryInterval time.Duration
-	// optsIncompleteBlockAccepted defines whether the node allows filling in incomplete block and issuing it for user.
-	optsIncompleteBlockAccepted bool
-	optsRateSetterEnabled       bool
+	AccountData AccountData
 }
 
-func NewBlockIssuer(t *testing.T, name string, keyManager *wallet.KeyManager, accountID iotago.AccountID, validator bool, opts ...options.Option[BlockIssuer]) *BlockIssuer {
-	priv, pub := keyManager.KeyPair()
+func NewBlockIssuer(t *testing.T, name string, keyManager *wallet.KeyManager, client Client, addressIndex uint32, accountID iotago.AccountID, validator bool, opts ...options.Option[BlockIssuer]) *BlockIssuer {
+	_, pub := keyManager.KeyPair(addressIndex)
 
 	if accountID == iotago.EmptyAccountID {
 		accountID = blake2b.Sum256(pub)
 	}
 	accountID.RegisterAlias(name)
 
+	accountAddress, ok := accountID.ToAddress().(*iotago.AccountAddress)
+	require.True(t, ok)
+
 	return options.Apply(&BlockIssuer{
-		Testing:                       t,
-		Name:                          name,
-		Validator:                     validator,
-		events:                        NewEvents(),
-		workerPool:                    workerpool.New("BlockIssuer"),
-		privateKey:                    priv,
-		PublicKey:                     pub,
-		AccountID:                     accountID,
-		optsIncompleteBlockAccepted:   false,
-		optsRateSetterEnabled:         false,
-		optsTipSelectionTimeout:       5 * time.Second,
-		optsTipSelectionRetryInterval: 200 * time.Millisecond,
+		Testing:                   t,
+		Name:                      name,
+		Validator:                 validator,
+		keyManager:                keyManager,
+		Client:                    client,
+		blockIssuanceResponseUsed: true,
+		AccountData: AccountData{
+			ID:           accountID,
+			AddressIndex: addressIndex,
+			Address:      accountAddress,
+		},
 	}, opts)
 }
 
 func (i *BlockIssuer) BlockIssuerKey() iotago.BlockIssuerKey {
-	return iotago.Ed25519PublicKeyHashBlockIssuerKeyFromPublicKey(hiveEd25519.PublicKey(i.PublicKey))
+	_, pub := i.keyManager.KeyPair()
+	return iotago.Ed25519PublicKeyHashBlockIssuerKeyFromPublicKey(hiveEd25519.PublicKey(pub))
 }
 
 func (i *BlockIssuer) BlockIssuerKeys() iotago.BlockIssuerKeys {
 	return iotago.NewBlockIssuerKeys(i.BlockIssuerKey())
 }
 
-// Shutdown shuts down the block issuer.
-func (i *BlockIssuer) Shutdown() {
-	i.workerPool.Shutdown()
-	i.workerPool.ShutdownComplete.Wait()
+func (i *BlockIssuer) Address() iotago.Address {
+	_, pub := i.keyManager.KeyPair()
```

Comment:
It looks weird that in the constructor you can pass an `addressIndex`, so the account address changes, but here it does not. Is this correct?